python/ray/autoscaler/_private/load_metrics.py

```diff
@@ -34,22 +38,51 @@ def update(self,
-        self.static_resources_by_ip[ip] = static_resources
-
-        # We are not guaranteed to have a corresponding dynamic resource for
-        # every static resource because dynamic resources are based on the
-        # available resources in the heartbeat, which does not exist if it is
-        # zero. Thus, we have to update dynamic resources here.
-        dynamic_resources_update = dynamic_resources.copy()
-        for resource_name, capacity in static_resources.items():
-            if resource_name not in dynamic_resources_update:
-                dynamic_resources_update[resource_name] = 0.0
-        self.dynamic_resources_by_ip[ip] = dynamic_resources_update
+        # If light heartbeat enabled, only resources changed will be received.
+        # We should update the changed part and compare static_resources with
+        # dynamic_resources using those updated.
+        if ray._config.light_heartbeat_enabled():
+            if ip not in self.static_resources_by_ip or \
+                    len(static_resources) > 0:
+                self.static_resources_by_ip[ip] = static_resources
+            if ip not in self.dynamic_resources_by_ip or \
+                    len(dynamic_resources) > 0:
+                if NODE_RESOURCES_TURNING_EMPTY in dynamic_resources:
+                    self.dynamic_resources_by_ip[ip] = {}
+                else:
+                    self.dynamic_resources_by_ip[ip] = dynamic_resources
+            if ip not in self.resource_load_by_ip or len(resource_load) > 0:
+                if NODE_RESOURCES_TURNING_EMPTY in resource_load:
+                    self.resource_load_by_ip[ip] = {}
+                else:
+                    self.resource_load_by_ip[ip] = resource_load
+            # We are not guaranteed to have a corresponding dynamic resource
+            # for every static resource because dynamic resources are based on
+            # the available resources in the heartbeat, which does not exist
+            # if it is zero. Thus, we have to update dynamic resources here.
+            dynamic_resources_update = self.dynamic_resources_by_ip[ip].copy()
+            for resource_name, capacity in \
+                    self.static_resources_by_ip[ip].items():
+                if resource_name not in dynamic_resources_update:
+                    dynamic_resources_update[resource_name] = 0.0
+            self.dynamic_resources_by_ip[ip] = dynamic_resources_update
+        else:
+            self.resource_load_by_ip[ip] = resource_load
+            self.static_resources_by_ip[ip] = static_resources
+            # We are not guaranteed to have a corresponding dynamic resource
+            # for every static resource because dynamic resources are based on
+            # the available resources in the heartbeat, which does not exist
+            # if it is zero. Thus, we have to update dynamic resources here.
+            dynamic_resources_update = dynamic_resources.copy()
```

Comment:
Is this the same in the `if` and `else`?

Reply:
Already refactored.